examples/which-problem.kl

```diff
@@ -7,12 +7,38 @@
         (>>= (which-problem)
           (lambda (prob)
             (case prob
-              [(declaration) (pure '(example (the (m) (m))))]
-              [(type)        (pure '(Bool))]
-              [(expression)  (pure '(true))]
-              [(pattern)     (pure '(unit))]))))]))
+              [(declaration)  (pure '(example (the (m) (m))))]
+              [(type)         (pure '(Bool))]
+              [(expression t) (pure '(true))]
+              [(pattern)      (pure '(unit))]))))]))
 
 (m)
 
 (example (case (unit)
            ((m) (true))))
+
+(define-macros
```

Comment:
This is so cool :smile:

Reply:
We could try to imlement something like Cryptol's zero operator using this. That'd be fun :-)